src/NuGet.Core/NuGet.Packaging/PackageCreation/Utility/PackageIdValidator.cs

```diff
@@ -19,6 +19,14 @@ public static class PackageIdValidator
             options: RegexOptions.IgnoreCase | RegexOptions.ExplicitCapture | RegexOptions.CultureInvariant,
             matchTimeout: TimeSpan.FromSeconds(10));
 
+        public static void ValidatePackageIdRegex(string packageId)
```

Comment:
>ValidatePackageIdRegex [](http://example.com/codeflow?start=27&length=22)

Since this method calls IsValidPackageId, I recommend this method is simply named ValidatePackageId. Another option is to rename IsValidPackageId to IsValidPackageIdRegex, which is less preferred.